bool ready
int8[] first
int8[] second
int8[] third
